DigitalLearningSolutions.Web.IntegrationTests/TestHelpers/TestUserApp.cs

```diff
@@ -0,0 +1,55 @@
+﻿namespace DigitalLearningSolutions.Web.IntegrationTests.TestHelpers
+{
+    using System;
+    using System.Linq;
+    using System.Security.Claims;
+    using DigitalLearningSolutions.Web.Helpers;
+    using Microsoft.AspNetCore.Authentication;
+    using Microsoft.AspNetCore.Builder;
+    using Microsoft.AspNetCore.Hosting;
+    using Microsoft.AspNetCore.Http;
+
+    public class TestUserAppStartupFilter : IStartupFilter
+    {
+        public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next)
+        {
+            return builder =>
+            {
+                next(builder);
+
+                builder.MapWhen(
+                    context => context.Request.Path.Value.StartsWith("/SetDelegateTestSession"),
+                    loginApp =>

```

Comment:
To check my understanding, this is effectively adding a new endpoint at `/SetDelegateTestSession` right?

Reply:
Yes it does, it sets the endpoint and acts like an auto login.